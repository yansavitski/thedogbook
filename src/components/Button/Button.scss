@import 'src/global';

.Button {
  align-items: center;
  background: $buttonColor;
  border: 1px solid $borderColor;
  border-radius: $borderRadius;
  color: $textColor;
  cursor: pointer;
  display: flex;
  font-size: 14px;
  padding: 10px;

  &:focus {
    outline: none;
  }

  &:hover {
    border: 1px solid $borderColorHover;
  }

  &_disabled {
    opacity: 0.6;
    pointer-events: none;
  }
}
