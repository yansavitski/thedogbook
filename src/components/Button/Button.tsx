import './Button.scss'

import classNames from 'classnames'
import React, { PointerEvent } from 'react'

interface ButtonProps {
  children: string | React.ReactElement
  className?: string
  disabled?: boolean
  onClick?: (event: PointerEvent<HTMLButtonElement>) => void
  type?: 'button' | 'submit'
}

export const Button: React.FunctionComponent<ButtonProps> = ({
  className,
  disabled,
  children,
  type = 'button',
  onClick,
}: ButtonProps) => {
  return (
    <button
      className={classNames(
        'Button',
        className,
        Boolean(disabled) && 'Button_disabled',
      )}
      type={type}
      onClick={onClick}
      disabled={disabled}
    >
      {children}
    </button>
  )
}
