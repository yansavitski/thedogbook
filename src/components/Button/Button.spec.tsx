import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'

import { Button } from './Button'

describe('<Button />', () => {
  it('renders correctly', () => {
    expect.assertions(1)
    const tree = shallow(<Button>button text</Button>)

    expect(toJson(tree)).toMatchSnapshot()
  })

  it('renders correctly with optional props', () => {
    expect.assertions(1)
    const tree = shallow(
      <Button className="btn-text" type="submit" disabled>
        button text
      </Button>,
    )

    expect(toJson(tree)).toMatchSnapshot()
  })
})
