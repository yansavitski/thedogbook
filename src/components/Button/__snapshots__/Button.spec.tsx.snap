// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<button
  className="Button"
  type="button"
>
  button text
</button>
`;

exports[`<Button /> renders correctly with optional props 1`] = `
<button
  className="Button btn-text Button_disabled"
  disabled={true}
  type="submit"
>
  button text
</button>
`;
